### PR TITLE
Update for Python 3

### PR DESCRIPTION
**Pull request by SuperQ**
* Update requirements to work with Python 3.
* Update controller use for new versions.
* Add Dockerfile

### CODE REVIEW

**Comment by SuperQ** [comment]
There's still a bunch of work to do, and it would be good to config the host from the config file. But this should get us enough to migrate from m4 to m5.

File: controller.py [hunk 1]
```diff
@@ -378,17 +378,17 @@ def index(page=1):
...
+        books = Book.query.order_by(Book.title.asc()).paginate(page=page, per_page=PAGINATE_BY_HOWMANY, error_out=False)
 
     return render_template('index.html', books=books, s=s)
 
 if __name__ == "__main__":
     # flask can execute arbitrary python if you do this.
     # app.run(host='0.0.0.0') # listens on all public IPs. 
 
-    app.run()
+    app.run(host='0.0.0.0')
```
Comment (jof) [inline controller.py]: Caddy fronts this with a reverse proxy

```suggestion
    app.run(host='127.0.0.1')
```

**Review by jof — APPROVED** [review]
Only a tiny nit on the bind IP; otherwise, looks more modern

File: controller.py (hunk 1 above)
Comment (SuperQ) [reply controller.py]: It needs to be `0.0.0.0` otherwise you can't port forward into docker.

Comment (SuperQ) [reply controller.py]: Added a section to the config file.

**Pull request merged**